projects/bp-gallery/src/components/common/ScrollContainer.tsx

```diff
@@ -1,17 +1,33 @@
-import React, { ReactComponentElement, useState } from 'react';
+import React, { ReactComponentElement, useEffect, useRef, useState } from 'react';
 import './ScrollContainer.scss';
+import { useHistory } from 'react-router-dom';
+import { History } from 'history';
 
 const ScrollContainer = ({
   children,
+  scrollTop,
 }: {
   children: (scrollPos: number, scrollHeight: number) => ReactComponentElement<any>;
+  scrollTop?: boolean;
 }) => {
   const [scrollPos, setScrollPos] = useState<number>(0);
   const [scrollHeight, setScrollHeight] = useState<number>(0);
+  const scrollDiv = useRef<HTMLDivElement>(null);
+  const history: History = useHistory();
+
+  //neccessary so that container scrolls to top when pressing on top bar logo on startpage
+  useEffect(() => {
+    if (scrollTop && history.location.state?.scrollTop) {
+      scrollDiv.current?.scrollTo(0, 0);
+      history.replace('', { showBack: false, scrollTop: false });
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  }, [scrollTop, history, history?.location?.state]);
```

Comment:
I'm not sure we need this. If I understand correctly, this only applies to startpage-to-startpage navigation, every other navigation works fine regarding scrolling, right? If so, I think I would be fine to just leave the scroll position as is when "navigating" from the startpage to the startpage, i. e. we could remove this code.

Reply:
done